Introductory Tests/Trajectory.py

```diff
@@ -0,0 +1,170 @@
+
+class Trajectory:
+    """
+    Holds all atributes, methods and functions related to integrating a trajectory
+    given an intitial state, mesh of a celestial body and the integration library
+    Desolver. The class also calls for files related to plotting the trajectory.
+    """
+
+    def __init__(self, body_density, final_time, start_time, time_step, algorithm, radius_bounding_sphere):
+        """ Setup udp attributes.
+
+        Args:
+            body_density (float): Mass density of body of interest
+            final_time (float): Final time for integration.
+            start_time (float): Start time for integration of trajectory (often zero)
+            time_step (float): Step size for integration. 
+            algorithm (int): User defined algorithm of choice
+            radius_bounding_sphere (float)_: Radius for the bounding sphere around mesh.
+        """
+        # Creating the mesh (TetGen)
+        self.body_mesh, self.mesh_vertices, self.mesh_faces, largest_body_protuberant = mesh_utility.create_mesh()
+
+        # Assertions:
+        assert body_density > 0
+        assert final_time > start_time
+        assert time_step <= (final_time - start_time)
+        assert radius_bounding_sphere > largest_body_protuberant
+
+        # Setup equations of motion class
+        self.eq_of_motion = EquationsOfMotion(self.mesh_vertices, self.mesh_faces, body_density)
+
+        # Additional hyperparameters
+        self.start_time = start_time
+        self.final_time = final_time
+        self.time_step = time_step
+        self.algorithm = algorithm
+        self.radius_bounding_sphere = radius_bounding_sphere
+
+
+    def integrate(self, x: np.ndarray) -> Union[np.ndarray, float, float]:
+        """compute_trajectory computes trajectory of satellite using numerical integation techniques 
+
+        Args:
+            x (np.ndarray): State vector containing values for position and velocity of satelite in three dimensions.
+
+        Returns:
+            trajectory_info (np.ndarray): Numpy array containing information on position and velocity at every time step (columnwise).
+            squared_altitudes (float): Sum of squared altitudes above origin for every position
+            collision_penalty (float): Penalty value given for the event of a collision with the celestial body.
+        """
+
+        # Integrate trajectory
+        initial_state = D.array(x)
+        trajectory = de.OdeSystem(
+            self.eq_of_motion.compute_motion, 
+            y0 = initial_state, 
+            dense_output = True, 
+            t = (self.start_time, self.final_time), 
+            dt = self.time_step, 
+            rtol = 1e-12, 
+            atol = 1e-12,
+            constants=dict(risk_zone_radius = self.radius_bounding_sphere)) #, mesh_vertices = self.mesh_vertices, mesh_faces = self.mesh_faces
+        trajectory.method = str(IntegrationScheme(self.algorithm).name)
+
+        point_is_inside_risk_zone.is_terminal = False
```

Comment:
Wait, actually shouldn't this be true? You want to abort if you collided after all?

Reply:
As of now, the event function is only used for detecting and storing the points on the trajectory that are inside the bounding sphere, or risk_zone, which we then use (after the integration) to check for possible collisions. If there is a collision, I have added a large penalty to mark it as infeasible, however, maybe it could be a good idea to also add a smaller penalty for all trajectories that goes inside the risk_zone (without necessarily colliding with the comet)?

In a previous version of this code, I included the collision check as part of the event function and set it to terminate if it found any collisions. However, running is_outside for each point that is inside the risk_zone individually took a lot of time, which made me reconsider the approach. By instead keeping a memory of points that were detected as inside the risk_zone, and then evaluating all such points for collision in one go reduced the time to less than a third compared to including it in the event function. However, this may also be a trade-off determined by the size of integration time.

Reply:
This is a bit tricky to decide. Because the idea of the event based thing is that you abort computing the trajectory the second a collision occurs. If the collision is early on, you save a lot of collision & point checks. Can keep as is for now but keep in mind this may be very suboptimal if there are a lot of collisions

Reply:
Then I will keep it as it is for now, however, another idea could also be to divide the integration time into some reasonable number of sub-intervals.